haskell/src/Lorentz/Contracts/Stablecoin.hs

```diff
@@ -62,26 +61,34 @@ metadataRegistryContractPath = "./test/resources/metadata.tz"
 ------------------------------------------------------------------
 -- Parameter
 
-type ConfigureMinterParam =
-  ( "minter" :! Address
-  , ( "current_minting_allowance" :! Maybe Natural
-    , "new_minting_allowance" :! Natural
-    )
-  )
+data ConfigureMinterParam = ConfigureMinterParam
+ { cmpMinter :: Address
+ , cmpCurrentMintingAllowance :: Maybe Natural
+ , cmpNewMintingAllowance :: Natural
+ }
+ deriving stock Show
+
+instance Buildable ConfigureMinterParam where
+  build = genericF
 
-pattern ConfigureMinterParams :: Address -> Maybe Natural -> Natural -> ConfigureMinterParam
-pattern ConfigureMinterParams addr cma nma <-
-  ( arg #minter -> addr
-  , (arg #current_minting_allowance -> cma, arg #new_minting_allowance -> nma))
-{-# COMPLETE ConfigureMinterParams #-}
+deriving anyclass instance IsoValue ConfigureMinterParam
+deriving anyclass instance HasAnnotation ConfigureMinterParam
+$(customGeneric "ConfigureMinterParam" $ withDepths
+    [ cstr @0
+      [ fld @1
+      , fld @2
+      , fld @2
```

Comment:
Apparently it's the same structure as the one that would be produced by vanilla `Generic` deriving. Actually I don't know what is better: explicitly write that we need this structure or prefer a shorter version that implicitly generates this structure. Both are somewhat fragile anyway :yao:

So I am suggesting this change, but I am not sure it's better than the current version, so feel free to ignore it.

Reply:
Yeah, I struggled with the same dilemma :sweat: 

I went for explicit-style, but I agree, both approaches are equally fragile. So now I'm starting to think we should go for the shorter version. I'll make this change.

Regarding the maintenance of these layouts: in #108, I changed [`parseStablecoinContract`](https://github.com/tqtezos/stablecoin/pull/108/files#diff-102dc12d0013cd28fb3bad312ffd8918L26) to return a typed contract instead of an untyped contract. `parseStablecoinContract` is used extensively in the test suites - so if a Generic layout goes out of sync, the tests are sure to catch this.

We can even add a compile-time check for this. With about ~5 lines of TH, we could typecheck the contract at compile-time and ensure the layouts match. What do you think? Or is this too much?

Reply:
> So now I'm starting to think we should go for the shorter version. I'll make this change.

Done

Reply:
Correction: 4 lines of TH :smile:  

```haskell
$(case readContract @(ToT Parameter) @(ToT Storage) stablecoinPath $(embedStringFile stablecoinPath) of
      Left e -> fail (pretty e)
      Right _ -> pure []
 )
```

Reply:
> Or is this too much?

I think 4 lines of TH is not too much, I'm ok with doing that :)

Reply:
Done